File: LeetCode/0198. House Robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int robbed = 0;
        int skipped = 0;

        for (int num : nums) {
            int previous_robbed = robbed;
            robbed = skipped + num;
            skipped = max(skipped, previous_robbed);
        }
        return max(robbed, skipped);
    }
};

File: LeetCode/0100-0199/0198. House Robber.java
class Solution {
    public int rob(int[] nums) {
        int skip = 0;
        int rob = 0;
        for (int num : nums) {
            int temp_skip = skip;
            skip = Math.max(skip, rob);
            rob = temp_skip + num;
        }
        return Math.max(skip, rob);
    }
}
